### PR TITLE
Add OSRM, Google, HERE and RoutingKit measure

### PR DESCRIPTION
**Pull request by larsbeck**
# Description

Adds measures for:
- OSRM
- Google
- HERE
- RoutingKit

### CODE REVIEW

File: route/google/doc.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+/*
+Package google provides functions for measuring distances and durations using
+the Google Distance Matrix API and polylines from Google Maps Distance API. A
+Google Maps client and request are required. The client uses an API key for
+authentication.
+Matrix API: At a minimum, the request requires the origins and destinations
+to estimate.
+Distance API: At minimum, the request requires the origin and destination. But
+it is recommended to pass in waypoints as an encoded as a polyline with "enc:"
```
Comment (nmisek) [inline route/google/doc.go]:
```suggestion
it is recommended to pass in waypoints encoded as a polyline with "enc:"
```

File: route/example_option_test.go [hunk 2]
```diff
@@ -490,6 +490,7 @@ func ExampleCapacity() {
 // Create routes to visit seven landmarks in Kyoto using two vehicles. In
 // addition precedences for stops are defined. The vehicles have no starting
 // locations and no maximum capacity that they can service.
+
```
Comment (sebastian-quintero) [inline route/example_option_test.go]: Plz remove all of these newlines.

File: route/example_router_test.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ import (
 )
 
 // Create routes to visit seven landmarks in Kyoto using two vehicles.
+
```
Comment (sebastian-quintero) [inline route/example_router_test.go]: Same here regarding the newline.

File: route/google/client.go [hunk 4]
```diff
@@ -0,0 +1,417 @@
+// © 2019-2022 nextmv.io inc. All rights reserved.
+// nextmv.io, inc. CONFIDENTIAL
+//
+// This file includes unpublished proprietary source code of nextmv.io, inc.
+// The copyright notice above does not evidence any actual or intended
+// publication of such source code. Disclosure of this source code or any
+// related proprietary information is strictly prohibited without the express
+// written permission of nextmv.io, inc.
```
Comment (sebastian-quintero) [inline route/google/client.go]: We need to remove these confidentiality headers in all of these files, as they are now under a different, open-source license.

**Review by sebastian-quintero — CHANGES_REQUESTED** [review]
This is a lot 😅 . Some comments regarding licenses and newlines.

**Comment by sebastian-quintero** [comment]
@larsbeck I took the liberty of helping you out with some modifications to the `go-buid-test.yml` and `go-lint.yml`. Since we have new go modules, we need to lint and test them. Standing at the root and running with `./...` doesn´t do the trick, we have to stand on each module to lint and test. I used a [job matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) to be able to do this without the wokflow file being too verbose.

This revealed many things that are still pending. I apologize for giving you more work 😅 .

**Review by sebastian-quintero — DISMISSED** [review]

**Review by sebastian-quintero — APPROVED** [review]

**Review by merschformann — APPROVED** [review]
:rocket:

**Pull request merged**